src/libsyntax/parse/parser.rs

```diff
@@ -5767,7 +5848,9 @@ impl<'a> Parser<'a> {
             (self.look_ahead(1, |t| t == &token::Pound || t == &token::Gt) ||
              self.look_ahead(1, |t| t.is_lifetime() || t.is_ident()) &&
                 self.look_ahead(2, |t| t == &token::Gt || t == &token::Comma ||
-                                       t == &token::Colon || t == &token::Eq))
+                                       t == &token::Colon || t == &token::Eq) ||
+             param && self.look_ahead(1, |t| t.is_keyword(keywords::Const) &&
+                self.look_ahead(2, |t| t.is_ident())))
```

Comment:
Not sure why `param` exists, it should always be `true`.
Also, `< const` is enough to identify a generic parameter, no need for look_ahead(2).

Reply:
> Not sure why `param` exists, it should always be `true`.

I have no idea now either; removing.